worker/src/RTC/SimulcastConsumer.cpp

```diff
@@ -821,13 +822,31 @@ namespace RTC
 			this->keyFrameForTsOffsetRequested = false;
 		}
 
+		if (!shouldSwitchCurrentSpatialLayer && this->checkingForOldPacketsInSpatialLayer)
```

Comment:
Does this logic work in case the SimulcastConsumer has been inactive for a while?

I'm worried about this sequence:
- ShouldSwitchCurrentSpatialLayer is true so we set checkingForOldPacketsInSpatialLayer
- After that the consumer is disabled (maybe because not enough bandwidth) for a while
- The consumer is resumed and the first sequence number after that is snReferenceSpatialLayer - 1 (because there was a rollover of sequence numbers).

Reply:
Yep, you were right. I'm now resetting `checkingForOldPacketsInSpatialLayer` to `false` when somebody call `UserOnResumed()`